plugin/src/main/java/org/opensearch/ml/action/prompt/TransportCreatePromptAction.java

```diff
@@ -0,0 +1,175 @@
+        log.info("prompt created, indexing into the prompt system index");
+        mlIndicesHandler.initMLPromptIndex(ActionListener.wrap(indexCreated -> {
+            if (!indexCreated) {
+                listener.onFailure(new RuntimeException("No response to create ML Prompt Index"));
+                return;
+            }
+
+            try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+                PutDataObjectRequest putRequest = buildPromptPutRequest(prompt);
+                sdkClient.putDataObjectAsync(putRequest).whenComplete((putResponse, throwable) -> {
+                    context.restore();
+                    handlePromptPutResponse(putResponse, throwable, listener);
+                });
+            }
+        }, e -> {
+            log.error("Failed to init ML prompt index", e);
+            listener.onFailure(e);
+        }));
+    }
+
+    /**
+     * Create a PutDataObjectRequest to store the prompt into the system index.
+     *
+     * @param prompt A prompt to be stored into the system index
+     * @return PutDataObjectRequest
+     */
+    private PutDataObjectRequest buildPromptPutRequest(MLPrompt prompt) {
+        return PutDataObjectRequest.builder().tenantId(prompt.getTenantId()).index(ML_PROMPT_INDEX).dataObject(prompt).build();
+    }
+
+    /**
+     * If the prompt is successfully stored into the system index, notify the listener with the MLCreatePromptResponse with prompt id.
+     * Otherwise, failure exception is notified to the listener.
+     *
+     * @param putResponse PutDataObjectResponse received from putDataObjectAsync op
+     * @param throwable Throwable received from putDataObjectAsync op. It is null if no exception thrown.
+     * @param listener ActionListener to be notified of the response
+     */
+    private void handlePromptPutResponse(
+        PutDataObjectResponse putResponse,
+        Throwable throwable,
+        ActionListener<MLCreatePromptResponse> listener
+    ) {
+        if (putResponse == null) {
+            Exception cause = SdkClientUtils.unwrapAndConvertToException(throwable);
+            handlePromptPutFailure(cause, listener);
+        }
+        try {
+            IndexResponse indexResponse = IndexResponse.fromXContent(putResponse.parser());
+            log.info("Prompt creation result: {}, prompt id: {}", indexResponse.getResult(), indexResponse.getId());
+            listener.onResponse(new MLCreatePromptResponse(indexResponse.getId()));
+        } catch (Exception e) {
+            handlePromptPutFailure(e, listener);
+        }
+    }
+
+    /**
+     * Notify the listener of the failure exception.
+     *
+     * @param cause The failure exception
+     * @param listener ActionListener to be notified of the response
+     */
+    private void handlePromptPutFailure(Exception cause, ActionListener<MLCreatePromptResponse> listener) {
```

Comment:
Lets take it one step further and add a argument that can hint us toward where this couldve went wrong for example 
```suggestion
    private void handlePromptPutFailure(Exception cause, ActionListener<MLCreatePromptResponse> listener, String likelyCause) {
       
       log.error("Failed to save ML Prompt : {}",likelyCause, cause);

```

Now when you invoke this method we can give the user a better hint as to where it couldnt pass. e.g. in your handlePromptPutResponse we can report that a likely cause would be that the prompt couldnt be parsed fromXContent.